### PR TITLE
Add test coverage for ResultsComparer and CLI helper paths

### PR DESCRIPTION
**Pull request by LoopedBard3**
## Summary

This expands automated coverage in the repo’s tooling-focused test surface without large production refactors.

### What changed

- Added a new `ResultsComparer.Tests` project covering:
  - file and directory input handling
  - JSON deserialization from file/stream
  - stats aggregation and output behavior
  - zip/tar.gz decompression flows
  - duplicate-result/version-selection behavior
  - CLI paths for invalid thresholds, missing inputs, and matrix output

- Expanded `BenchmarkDotNet.Extensions` command-line option tests to cover:
  - string parameter parsing
  - boolean switch parsing
  - missing/invalid integer argument paths
  - absent-switch behavior

- Improved `Startup.Tests` reliability by skipping ETW-dependent Windows tests when not running elevated, instead of aborting the run

### Small correctness fixes found while adding coverage

The new tests exposed a few edge-case bugs, which are fixed in this PR:

- `CommandLineOptions.ParseAndRemoveStringsParameter` now handles missing switches safely
- `ResultsComparer.Program` now tolerates a missing optional `--filter`
- `TwoInputsComparer` now treats `EquivalenceTestConclusion.Base` as a no-diff case
- `ResultsComparer.Data` moniker parsing is now null-safe for additional archive path shapes

## Validation

Validated the affected suites in the available environment:

- `ResultsComparer.Tests` passing
- targeted `BenchmarkDotNet.Extensions` command-line tests passing
- `Startup.Tests` now skip environment-dependent cases cleanly

> **Note:** test projects are kept at `net11.0`; full execution of those targets requires a local/CI environment with the .NET 11 SDK installed.

### CODE REVIEW

File: src/tools/ResultsComparer.Tests/ProgramTests.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            tempDir.Delete(recursive: true);
+        }
+    }
+
+    private static string InvokeProgram(string[] args)
+    {
+        using var writer = new StringWriter();
+        var originalOut = Console.Out;
+        Console.SetOut(writer);
+        try
```
[the remark this answers is not in the document]
Comment (LoopedBard3) [reply src/tools/ResultsComparer.Tests/ProgramTests.cs:146]: This was added to Collection("Console output") which has parallelization disabled.

File: src/tools/ResultsComparer.Tests/StatsTests.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        stats.Record(EquivalenceTestConclusion.Same, environment, benchmark);
+        stats.Record(EquivalenceTestConclusion.Faster, environment, benchmark);
+        stats.Record(EquivalenceTestConclusion.Slower, environment, benchmark);
+        stats.Record(EquivalenceTestConclusion.Unknown, environment, benchmark);
+        stats.Record(global::ResultsComparer.Stats.Noise, environment, benchmark);
+
+        using var writer = new StringWriter();
+        var originalOut = Console.Out;
+        Console.SetOut(writer);
+        try
```
[the remark this answers is not in the document]
Comment (LoopedBard3) [reply src/tools/ResultsComparer.Tests/StatsTests.cs:49]: This was added to Collection("Console output") which has parallelization disabled.

**Review by DrewScoggins — APPROVED** [review]
LGTM
